core/trino-spi/src/main/java/io/trino/spi/type/BooleanType.java

```diff
@@ -41,6 +44,23 @@
 
     public static final BooleanType BOOLEAN = new BooleanType();
 
+    /**
+     * This method signifies a contract to callers that as an optimization, they can encode BooleanType blocks as a byte[] directly
+     * and potentially bypass the BlockBuilder / BooleanType abstraction in the name of efficiency. If in the future BooleanType
+     * encoding changes such that {@link ByteArrayBlock} is not always a valid or efficient representation, then this method must be
+     * removed and any usages changed
+     */
+    public static Block wrapByteArrayAsBooleanBlockWithoutNulls(byte[] booleansAsBytes)
```

Comment:
nit: we probably should have similar methods for other types and have readers use them. cc @skrzypo987